Libplanet/Net/TxCompletion.cs

```diff
@@ -0,0 +1,191 @@
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    _txSyncTasks.TryRemove(peer, out _);
+                    throw new TaskCanceledException();
+                }
+
+                try
+                {
+                    HashSet<TxId> txIds = GetRequiredTxIds(_requiredTxIds[peer]);
+                    _requiredTxIds[peer] = new ConcurrentBag<TxId>();
+                    var txs = new HashSet<Transaction<TAction>>(
+                        await _txFetcher(
+                                peer,
+                                txIds,
+                                cancellationToken)
+                            .ToListAsync(cancellationToken)
+                            .AsTask());
+
+                    txs = new HashSet<Transaction<TAction>>(
+                        txs.Where(
+                            tx =>
+                            {
+                                if (_blockChain.Policy.ValidateNextBlockTx(_blockChain, tx) is null)
+                                {
+                                    return true;
+                                }
+
+                                _logger.Debug(
+                                    "Received transaction {TxId} will not be staged " +
+                                    "since it does not follows policy.",
+                                    tx.Id);
+                                _blockChain.StagePolicy.Ignore(_blockChain, tx.Id);
+                                return false;
+                            }));
+
+                    foreach (var tx in txs)
+                    {
+                        try
+                        {
+                            _blockChain.StageTransaction(tx);
+                        }
+                        catch (InvalidTxException ite)
+                        {
+                            _logger.Error(
+                                ite,
+                                "{TxId} will not be staged since it is invalid.",
+                                tx.Id);
+                        }
+                    }
+
+                    if (txs.Any())
+                    {
+                        TxReceived.Set();
+                        _logger.Debug(
+                            "Txs staged successfully: {@TxIds}",
+                            txs.Select(tx => tx.Id.ToString()));
+
+                        // TODO: txs includes transaction which were ignored due to its nonce,
+                        // which should not be re-broadcasted.
+                        _txBroadcaster(peer, txs);
+                    }
+                    else
+                    {
+                        _logger.Information(
+                            "Failed to get transactions to stage: {@TxIds}",
+                            _requiredTxIds[peer].Select(txId => txId.ToString()));
+                    }
+                }
+                finally
+                {
+                }
```

Comment:
IMO, an empty `finally` isn't needed usually.

Reply:
Without finally, the analyzer throws error.